truffle-box/react-box
# Web3 v1.0

This PR seeks to overhaul the Truffle box in many ways, read on to find out more.

# Done

The following items have already been implemented in this PR.

## Frontend client

- Replaced old React frontend with a new one located in its own separate folder `/client`.
  - Uses **non-ejected** Create-React-App as its base.
- Uses Web3 `v1.0-beta.35` (note: currently, only for the React app).
- Uses custom script in `getContractInstance.js` to replace `truffle-contract` dependency (this can change).
- Uses a `link-contracts` npm script to create a symlink in the `src` folder
  -  This links to the json contract build artifacts folder in the outer Truffle project 

## Truffle project

- Update `truffle-box.json` to use new scripts when unboxing.
- Update test code to use Async/Await Javascript pattern.
- Bump Solidity pragma versions to `^0.4.24`.

# How to test (i.e. try it out)

1. Checkout the branch:

   ```
   git checkout web3-v1.0
   ```

2. Run `truffle develop` and then run tests, compile, and migrate:

   ```
   truffle develop
   truffle(develop)> test
   truffle(develop)> compile
   truffle(develop)> migrate --reset
   ```

3. In a new terminal, navigate to the `client` folder, install dependencies, and run the link contracts script:

   ```bash
   cd client
   npm install
   npm run link-contracts
   ```

4. Start the frontend client with `npm start`.

5. Make sure you are using a browser where MetaMask is disabled (we will add support for this later, see above) and navigate to `http://locahost:3000` in your browser.

6. Observe that the stored value is 5.
